services/community-site/src/pages/RunNode/Screen/UpdateValidator.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from 'react';
+import { Text, Button, InputField, Modal, useInterval } from '@taraxa_project/taraxa-ui';
+import { Validator } from '../../../interfaces/Validator';
+// import useValidators from 'services/community-site/src/services/useValidators';
+import Title from '../../../components/Title/Title';
+import CloseIcon from '../../../assets/icons/close';
+import UpdateCommission, { VALIDATOR_COMMISSION_CHANGE_FREQUENCY } from '../Modal/UpdateCommission';
+import useValidators from '../../../services/useValidators';
+import useChain from '../../../services/useChain';
+
+const MAX_ENDPOINT_LENGTH = 50;
+
+const MAX_DESCRIPTION_LENGTH = 100;
+
+interface UpdateValidatorProps {
+  closeEditValidator: (refreshValidators: boolean) => void;
+  validator: Validator;
+}
+
+const UpdateValidator = ({ closeEditValidator, validator }: UpdateValidatorProps) => {
+  const { setValidatorInfo } = useValidators();
+  const { provider } = useChain();
+
+  const [description, setDescription] = useState(validator.description || '');
+  const [descriptionError, setDescriptionError] = useState('');
+  const [endpoint, setEndpoint] = useState(validator.endpoint || '');
+  const [endpointError, setEndpointError] = useState('');
+  const [currentBlock, setCurrentBlock] = useState(0);
+  const [isUpdatingCommission, setIsUpdatingCommission] = useState(false);
+
+  const getCurrentBlock = async () => {
+    if (provider) {
+      setCurrentBlock(await provider.getBlockNumber());
+    }
+  };
+
+  useInterval(async () => {
+    getCurrentBlock();
+  }, 5000);
+
+  const canChangeCommission =
+    !currentBlock ||
```

Comment:
Shouldn't this be the other way around?

Reply:
I mean, until the currentBlock isn't fetched the second part is always false so it's kinda redundant

Reply:
Don't understand. So the button should be enabled if the current block isn't fetched yet?

Reply:
If we enable it before the blocks being fetched we introduce uncertainty, I'd vote for keeping it disabled and letting the user trigger the flow only if they can really change regardless of the delay on the frontend.